frontend/src/components/VariantListPage/VariantsTable.tsx

```diff
@@ -747,12 +751,21 @@ const VariantsTable: FC<VariantsTableProps> = ({
               alignItems: "center",
             }}
           >
-            {column.render(
-              rowData,
-              variantList,
-              variantNotes,
-              onEditVariantNote,
-              userCanEdit
+            {column.key === "variant_id" ? (
+              <Highlighter
+                highlightClassName="highlight"
+                searchWords={[searchText]}
+                autoEscape
+                textToHighlight={rowData.id}
+              />
+            ) : (
+              column.render(
+                rowData,
+                variantList,
+                variantNotes,
+                onEditVariantNote,
+                userCanEdit
+              )
             )}
```

Comment:
**issue**: I like the functionality of getting the highlighting working here, though removing the call to the column's `render` in this case removed the links to gnomAD (I actually only noticed this at the end of my review because the snapshots changed and removed the links -- neat!)

**suggestion**: I think we can have both highlighting, and the link, if we move the highlight logic into the render function for the `variant_id` column, something like this should do the trick:

These lines return to what they were before with no ternary, but now include a sixth argument:
``` 
{column.render(
	rowData,
	variantList,
	variantNotes,
	onEditVariantNote,
	userCanEdit,
	searchText,   <-- pass this to get access to it in the render function
)}
```

Then, we change the column def to add the `searchText` and a default arg to the render function (line ~128):

```
interface ColumnDef {
  key: string;
  heading: string;
  isNumeric?: boolean;
  width: number;
  sortKey?: (
    variant: Variant,
    variantList: VariantList
  ) => string | number | (string | number)[];
  render: (
    variant: Variant,
    variantList: VariantList,
    variantNotes: Record<VariantId, string>,
    onEditVariantNote: (variantId: VariantId, note: string) => void,
    userCanEdit: boolean,
    searchText: string,  <-- take in another arg
  ) =>
    | JSX.Element
    | string
    | (JSX.Element | string)[]
    | null
    | undefined
    | false
    | "";       <-- add a default arg
}
```

Then finally, we can use this `searchText` passed to us in only the `variant_id` column def to have highlighting and linking (line ~156):

```
{
    key: "variant_id",
    heading: "Variant ID",
    width: 190,
    sortKey: (variant) => {
      const [chrom, pos, ref, alt] = variant.id.split("-");
      return [chrom, Number(pos), ref, alt];
    },
    render: (
      variant,
      variantList,
      _variantNotes,        <-- take in parameters in order, use _ b/c we don't use
      _onEditVariantNotes,  <--   same as above
      _userCanEdit,         <--   same as above
      searchText,           <-- get access to search text in render func
    ) => {
      const gnomadVersion = variantList.metadata.gnomad_version;
      const gnomadDataset = {
        "2.1.1": "gnomad_r2_1",
        "3.1.2": "gnomad_r3",
        "4.0.0": "gnomad_r4",
        "4.1.0": "gnomad_r4",
      }[gnomadVersion];

      const gnomadSVDataset = {
        "2.1.1": "gnomad_sv_r2_1",
        "3.1.2": "gnomad_sv_r4",
        "4.0.0": "gnomad_sv_r4",
        "4.1.0": "gnomad_sv_r4",
      }[gnomadVersion];

      const dataset = isStructuralVariantId(variant.id, gnomadVersion)
        ? gnomadSVDataset
        : gnomadDataset;
      const variantId = isStructuralVariantId(variant.id, gnomadVersion)
        ? variant.id.toUpperCase()
        : variant.id;

      return (
        <Cell maxWidth={190}>
          <Link
            href={`https://gnomad.broadinstitute.org/variant/${variantId}?dataset=${dataset}`}
            isExternal
            target="_blank"
          >

			// here, instead of simply returning {variant.id} inside the link
			// we can return the hightlighted HTML, inside the link still

            <Highlighter                   
              highlightClassName="highlight"
              searchWords={[searchText]}
              autoEscape
              textToHighlight={variant.id}
            />
          </Link>
        </Cell>
      );
```

e.g.:
![Screenshot 2024-12-12 at 15 40 00](https://github.com/user-attachments/assets/d371d648-d5a6-40e8-899a-86a11c0c6198)